src/sentry/grouping/enhancer/actions.py

```diff
@@ -28,27 +30,29 @@ class Action:
     _is_modifier: bool
     _is_updater: bool
 
-    def apply_modifications_to_frame(self, frames, match_frames, idx, rule=None):
+    def apply_modifications_to_frame(self, frames, match_frames, idx):
         pass
 
-    def update_frame_components_contributions(self, components, frames, idx, rule=None):
+    def update_frame_components_contributions(
+        self, components: Any, frames: Sequence[dict[str, Any]], idx: Any, rule: Any = None
```

Comment:
I have lots of `Any` in this file since it was getting really complicated. Eventually we will clean these up.

Reply:
Hmmm - I've been thinking of `Any` as a little TODO flag for the future, as it's harder to search for missing types than that string, when we decide we want to go through and clean things up.

Reply:
`mypy --disallow-incomplete-defs` or `mypy --disallow-untyped-defs` is the intended mechanism for finding incomplete types

Reply:
Ah, TIL!